lib/audit_logging/audit_log.ex

```diff
@@ -0,0 +1,79 @@
+defmodule PrimaExLogger.AuditLogging.AuditLog do
+  @moduledoc """
+  Struct representation of and Audit log
+  """
+
+  @audit_log_scope "auditLog"
+
+  @derive Jason.Encoder
+  @enforce_keys [:actor, :event_name, :message, :timestamp]
+  defstruct [
+    :actor,
+    :event_name,
+    :message,
+    :timestamp,
+    :created_at,
+    :metadata,
+    :target,
+    :http,
+    :runtime,
+    scope: @audit_log_scope
+  ]
+
+  def created_at(log, created_at), do: %{log | created_at: created_at}
+  def metadata(log, metadata), do: %{log | metadata: metadata}
+  def target(log, target), do: %{log | target: target}
+  def http(log, http), do: %{log | http: http}
+  def runtime(log, runtime), do: %{log | runtime: runtime}
```

Comment:
it would be helpful to have typespecs here, I guess `runtime` should receive a `Runtime` struct and the same applies to `http`

Reply:
Good point! Added them, not sure if I'm using the right syntax in the module funcs i.e. I don't know if I should refer to the module itself as `__MODULE__.t()` or do as I'm doing

Reply:
Or even just `t()`...? 🤔